services/mirror/mirror_push.go

```diff
@@ -156,11 +157,33 @@ func runPushSync(ctx context.Context, m *repo_model.PushMirror) error {
 
 		log.Trace("Pushing %s mirror[%d] remote %s", path, m.ID, m.RemoteName)
 
+		var initArgs []string
+		if m.PublicKey != "" {
+			f, err := os.CreateTemp(os.TempDir(), m.RemoteName)
+			if err != nil {
+				log.Error("CreateTemp: %v", err)
+				return errors.New("unexpected error")
+			}
+			defer func() {
+				f.Close()
+				if err := os.Remove(f.Name()); err != nil {
+					log.Error("os.Remove: %v", err)
+				}
+			}()
+
+			if _, err := f.Write([]byte(m.PrivateKey)); err != nil {
+				log.Error("f.Write: %v", err)
+				return errors.New("unexpected error")
+			}
+
+			initArgs = append(initArgs, "-c", fmt.Sprintf("core.sshcommand=ssh -i %q -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no", f.Name()))
```

Comment:
Perhaps one or two comments would be fine here to explain why you are creating a temp file with the private key as value.
I don't want to always have to read through 20 lines of code, only to understand its purpose…